File: py/solns/naryTreePostorderTraversal/naryTreePostorderTraversal.py
class Node:
    def __init__(self,val=None,children=None):
        self.val = val
        self.children = children if children is not None else []

class Solution:
    @staticmethod
    def naive(root):
        res = []
        def postOrder(tree):
            if tree==None: return 
            for child in tree.children:
                postOrder(child)
            res.append(tree.val)
        postOrder(root)
        return res

File: cpp/example/src/NaryTreePostorderTraversal/naryTreePostorderTraversal.cpp
#include "naryTreePostorderTraversal.h"

void naryTreePostorderTraversal::recursive(Node* tree){
    if (tree==nullptr) return;
    for (auto child: tree->children)
        recursive(child);
    this->res.push_back(tree->val);
}

vector<int> naryTreePostorderTraversal::naive(Node* root){
    recursive(root);
    return this->res;
}
